backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -713,6 +713,24 @@ internal class FunctionGenerationContext(val function: LLVMValueRef,
         return exceptionPtr
     }
 
+    private fun createForeignException(landingpadResult: LLVMValueRef, exceptionHandler: ExceptionHandler): LLVMValueRef {
+        val exceptionRecord = extractValue(landingpadResult, 0, "er")
+
+        // __cxa_begin_catch returns pointer to C++ exception object.
+        val exceptionRawPtr = call(context.llvm.cxaBeginCatchFunction, listOf(exceptionRecord))
+
+        // ObjC expects NSException** here, so do dereference
```

Comment:
What happens with other C++ exceptions here?

Reply:
C++ exception now safely passed and can be caught in other C++ frame.  Control flow reaches `resume`, as we don't have universal `clause null` anymore ( aka `catch (...)` ) .

Reply:
I'm not sure that it would work this way. Do Kotlin `catch` blocks safely pass C++ exceptions?
Also passing C++ exceptions through Kotlin code like this is not desirable at the moment, as we discussed earlier.